File: java/src/test/java/codingdojo/CompensationCalculatorTest.java
package codingdojo;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CompensationCalculatorTest {
    @Test
    void calculateOvertime() {
        Integer[] assignmentHours = {0, 5, 6};
        Integer[] overtimeTotalHours = {0, 1, 10, 12, 17};
        CombinationApprovals.verifyAllCombinations(this::doCalculateOvertime,
                assignmentHours,
                overtimeTotalHours,
                new Boolean[] {true, false},
                new Boolean[] {true, false},
                new Boolean[] {true, false},
                new Boolean[] {true, false},
                new Boolean[] {true, false}
        );

    }

    private Overtime doCalculateOvertime(int assignmentHours, int overtimeTotalHours, boolean isUnionized, boolean watcode, boolean z3, boolean foreign, boolean hbmo) {
        Assignment assignment = new Assignment(isUnionized, Duration.ofHours(assignmentHours));
        Briefing briefing = new Briefing(watcode, z3, foreign, hbmo);
        BigDecimal overtimeTotal = BigDecimal.valueOf(overtimeTotalHours);
        return CompensationCalculator.calculateOvertime(overtimeTotal, assignment, briefing);
    }

}

File: csharp/OvertimeCalculation/OvertimeCalculationTest/CompensationCalculatorTest.cs
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using OvertimeCalculation;

namespace OvertimeCalculationTest;

[UseReporter(typeof(QuietReporter))]
public class CompensationCalculatorTest
{
    [Test]
    public void CalculateOvertime()
    {
        int[] assignmentHours = { 0, 5, 6 };
        int[] overtimeTotalHours = { 0, 1, 10, 12, 17 };
        CombinationApprovals.VerifyAllCombinations(DoCalculateOvertime,
            assignmentHours,
            overtimeTotalHours,
            new[] { true, false },
            new[] { true, false },
            new[] { true, false },
            new[] { true, false },
            new[] { true, false }
        );
    }

    private Overtime DoCalculateOvertime(int assignmentHours, int overtimeTotalHours, bool isUnionized, bool watcode, bool z3, bool foreign, bool hbmo)
    {
        var assignment = new Assignment(isUnionized, TimeSpan.FromHours(assignmentHours));
        var briefing = new Briefing(watcode, z3, foreign, hbmo);
        decimal overtimeTotal = overtimeTotalHours;
        return CompensationCalculator.CalculateOvertime(overtimeTotal, assignment, briefing);
    }
}
